watertap/core/util/scaling.py

```diff
@@ -0,0 +1,149 @@
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+This module contains utility functions for testing the scaling of WaterTAP models.
+"""
+
+import pyomo.environ as pyo
+from pyomo.common.collections import ComponentSet
+
+import idaes.logger
+from idaes.core.solvers import get_solver
+from idaes.core.util.scaling import (
+    calculate_scaling_factors,
+    badly_scaled_var_generator,
+    get_scaling_factor,
+    unset_scaling_factor,
+)
+
+solver = get_solver()
+
+__author__ = "Hunter Barber"
+
+
+def variable_sens_generator(blk, lb_scale=1e-2, ub_scale=1e2, tol=1e3, zero=1e-10):
+    """
+    Generator which varies the magnitude of inlet flow(s) to some lower and upper specification
+        for a square and solved flowsheet. Previous scaling factors in the model are cleared and default
+        scaling factors are adjusted to account for the new flow magnitude and again propagated through
+        the variables on initialization. The model is re-solved and tested for: (i) are any variables
+        poorly scaled at different process scales, (ii) are any scaled variables with static scaling
+        factors proportional the process flow and therefore candidate for implementing dynamic scaling
+        factors, (iii) do any variables touch their bounds between process scales that were not
+        explicitly tested in other test frames, leading to non-optimal solves.
+
+    Keyword Arguments:
+        blk : pyomo ConcreteModel block which holds the flowsheet to be tested
+        lb_scale : the lower bound scale which adjusts the magnitude for the
+            fixed inlet flows of the process (default = 1e-2)
+        ub_scale : the upper bound scale which adjusts the magnitude for the
+            fixed inlet flows of the process (default = 1e2)
+        tol : tolerance for the change in magnitude of scaled variables between
+            the lower and upper bounds (default = 1e3), should always be smaller
+            than the difference in magnitude of lb_scale and up_scale
+        zero: magnitude that is considered to be zero, scaled variables with a value of
+            zero are okay, and not reported (default = 1e-10)
+
+    Yields:
+        string denoting what step failed, variable name as string,
+            sensitivity of sv between lower and upper solutions
+    """
+
+    # TODO: Check DOF is 0
+
+    sv_hist = {}
+
+    for scale in [lb_scale, ub_scale]:
+
+        temp_blk = (
+            blk.clone()
+        )  # clone flowsheet to re-solve at conditions supplied by scale
```

Comment:
```suggestion
        # clone flowsheet to re-solve at conditions supplied by scale
        temp_blk = blk.clone()
```

This is a really minor nit, but we need to be putting comments *above* the line of code they apply to. Black comes along as does this to it, which makes things harder to read.